### PR TITLE
[CPU] Quantize: Inverted input intervals support

### PR DESCRIPTION
**Pull request by vzinovie**
issue: 51673
validation: developer-services/job/accuracy/1661/
reference: developer-services/job/accuracy/1662/

### CODE REVIEW

File: inference-engine/src/transformations/src/transformations/op_conversions/fq_decomposition.cpp [hunk 1]
```diff
@@ -62,10 +62,15 @@ ngraph::pass::FakeQuantizeDecomposition::FakeQuantizeDecomposition() {
             decomp_ops.push_back(data.get_node_shared_ptr());
         }
 
+        // compute min and max intervals bounds to support inverted intervals
```
[the remark this answers is not in the document]
Comment (vzinovie) [reply inference-engine/src/transformations/src/transformations/op_conversions/fq_decomposition.cpp:65]: @mandrono Yes, this behavior is correct for inverted intervals. Low values are mapped on high values and vice-versa.

**Comment by mandrono** [comment]
May I ask you to remove check ```il > ih``` looks like it's not valid anymore?
https://github.com/openvinotoolkit/openvino/blob/f0705ecce9bf7e75fa05defbb20a5e1d0a148df1/inference-engine/src/mkldnn_plugin/nodes/mkldnn_quantize_node.cpp#L1029-L1034

**Review by antonvor — APPROVED** [review]
Looks good for me but correct comments above

**Review by mikhail-treskin — APPROVED** [review]

File: docs/ops/quantization/FakeQuantize_1.md [hunk 2]
```diff
@@ -13,13 +13,13 @@
...
 
 ```python
-if x <= min(input_low, input_high):
-    output = output_low
-elif x > max(input_low, input_high):
-    output = output_high
-else:
-    # input_low < x <= input_high
-    output = round((x - input_low) / (input_high - input_low) * (levels-1)) / (levels-1) * (output_high - output_low) + output_low
+input_min = min(input_low, input_high)
```
Comment (mandrono) [inline docs/ops/quantization/FakeQuantize_1.md:16]: what about case then ```input_low == input_high``` ?

Comment (vzinovie) [reply docs/ops/quantization/FakeQuantize_1.md:16]: @mandrono Good point! Corrected. Thank you!

File: docs/ops/quantization/FakeQuantize_1.md [hunk 3]
```diff
@@ -13,13 +13,14 @@
 Each element of the output is defined as the result of the following expression:
 
 ```python
-if x <= min(input_low, input_high):
-    output = output_low
-elif x > max(input_low, input_high):
-    output = output_high
+if input_low != input_high:
```
Comment (jane-intel) [inline docs/ops/quantization/FakeQuantize_1.md:16]: Are these changes backward compatible? Seems like not really. Could you provide detailed explanation what has changed and wny?

Comment (vzinovie) [reply docs/ops/quantization/FakeQuantize_1.md:16]: @jane-intel Previous spec behavior was not corrected for inverted intervals + values that out of bounds. To handle such values correctly for all 4 cases and not create multiple conditions (inverted/plain; in/out), I suggest just values right and pass them through a general formula.
If we are talking about branch where `input_low == input_high`, behavior is the same as on previous spec.

Comment (slyalin) [reply docs/ops/quantization/FakeQuantize_1.md:16]: Why do we need to handle inverted intervals in a special way? What does it mean that it was incorrect for out of bounds???

**Pull request closed without merging**